services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -0,0 +1,164 @@
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destinationConfig interface{}) (pubsub.Client, error) {
+	var config Config
+	ctx := context.Background()
+	jsonConfig, err := json.Marshal(destinationConfig)
+	if err != nil {
+		return pubsub.Client{}, fmt.Errorf("[GooglePubSub] Error while marshalling destination config :: %w", err)
+	}
+	err = json.Unmarshal(jsonConfig, &config)
+	if err != nil {
+		return pubsub.Client{}, fmt.Errorf("[GooglePubSub] error  :: error in GooglePubSub while unmarshelling destination config:: %w", err)
+	}
+	var client *pubsub.Client
+	if config.Credentials != "" && config.ProjectId != "" {
+		client, err = pubsub.NewClient(ctx, config.ProjectId, option.WithCredentialsJSON([]byte(config.Credentials)))
+	}
+	if err != nil {
+		return pubsub.Client{}, err
+	}
+	return *client, nil
+}
+func Produce(jsonData json.RawMessage, producer interface{}, destConfig interface{}) (statusCode int, respStatus string, responseMessage string) {
+	parsedJSON := gjson.ParseBytes(jsonData)
+	pbs, ok := producer.(pubsub.Client)
+	ctx := context.Background()
+	if !ok {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: Could not create producer"
+		return 400, respStatus, responseMessage
+	}
+	var config Config
+	jsonConfig, err := json.Marshal(destConfig)
+	if err != nil {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: " + err.Error()
+		logger.Errorf("[GooglePubSub] error  :: %w", err)
+		statusCode := 400
+		return statusCode, respStatus, responseMessage
+	}
+	err = json.Unmarshal(jsonConfig, &config)
+	if err != nil {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error  :: " + err.Error()
+		logger.Errorf("[GooglePubSub] error  :: %w", err)
+		statusCode := 400
+		return statusCode, respStatus, responseMessage
+	}
+	var data interface{}
+	if parsedJSON.Get("message").Value() != nil {
+		data = parsedJSON.Get("message").Value()
+	} else {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: message from payload not found"
+		return 400, respStatus, responseMessage
+	}
+	value, err := json.Marshal(data)
+
+	if err != nil {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error  :: " + err.Error()
+		logger.Errorf("[GooglePubSub] error  :: %w", err)
+		statusCode := 400
+		return statusCode, respStatus, responseMessage
+	}
+	if parsedJSON.Get("topicId").Value() != nil {
+		topicIdString, ok := parsedJSON.Get("topicId").Value().(string)
+		if !ok {
+			respStatus = "Failure"
+			responseMessage = "[GooglePubSub] error :: Could not parse topic id to string"
+			logger.Error(responseMessage)
+			statusCode := 400
+			return statusCode, respStatus, responseMessage
+		}
+		if topicIdString == "" {
+			respStatus = "Failure"
+			responseMessage = "[GooglePubSub] error :: empty topic id string"
+			return 400, respStatus, responseMessage
+		}
+		topic := pbs.Topic(topicIdString)
```

Comment:
// If a Topic's Publish method is called, it has background goroutines
// associated with it. Clean them up by calling Topic.Stop.
//
// Avoid creating many Topic instances if you use them to publish.
func (c *Client) Topic(id string) *Topic {

Referring to the method's documentation, there are following issues:
1. We are not calling topic.Stop(), which will leave behind goroutines. Verify this by looking at dataplane health endpoint, which contains goroutine count. 
2. I think we should reuse topic objects instead of creating them again and again. And stop topics whenever producer is closed.

Reply:
done